apollo-compiler/src/test/kotlin/com/apollographql/apollo3/compiler/FieldsOnDisjointTypesMustMergeTest.kt

```diff
@@ -0,0 +1,33 @@
+package com.apollographql.apollo3.compiler
+
+import com.apollographql.apollo3.ast.parseAsGQLDocument
+import com.apollographql.apollo3.ast.validateAsExecutable
+import com.apollographql.apollo3.compiler.TestUtils.checkExpected
+import com.apollographql.apollo3.compiler.TestUtils.findSchema
+import com.apollographql.apollo3.compiler.TestUtils.serialize
+import okio.buffer
+import okio.source
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import java.io.File
+
+class FieldsOnDisjointTypesMustMergeTest {
+  private val graphQLFile = File("src/test/validation/operation/fields_on_disjoint_types_must_merge/different_shapes.graphql")
+
+  @Test
+  fun testEnableFieldsOnDisjointTypesMustMerge() = checkExpected(graphQLFile) { schema ->
```

Comment:
This test is run twice:
- once here
- once in `ValidationTest`

Generally speaking, `src/test/validation/` is "owned" by `ValidationTest` so I would just use the `ValidationTest` infrastructure. Something like this (assuming you rename `different_shapes` to `merging_allowed`:

```kotlin
        val mustMerge = graphQLFile.name != "merging_allowed.graphql"
        parseResult.valueAssertNoErrors().validateAsExecutable(schema = schema!!, fieldsOnDisjointTypesMustMerge = mustMerge).issues + if (graphQLFile.name == "capitalized_fields_disallowed.graphql") {
```

Reply:
I updated the test - added `merging_allowed` which is the same as `different_shapes` but it's used to validate the case where `fieldsOnDisjointTypesMustMerge` is `false`. let me know if I understand your suggestion correctly

Reply:
Yep, all good, thanks!